# shell script to statistic depth
```shell
target=NC_007605.bed
echo -e "SampleID\tCoverage_of_target_region\tAverage_sequencing_depth_on_target\tFraction_of_target_covered_with_at_least_5x\tFraction_of_target_covered_with_at_least_10x" >> ${id}.coverage
sample=${id}.bam;
Out=${id}.d.all.txt
bedtools coverage -a ${target} -b ${sample} -d >${Out};
target_length=$(wc -l <${Out});
av=$(awk '{sum=sum+$5} END{print sum/"'${target_length}'"}' "${Out}");
C1=$(awk '$5>0' "${Out}"|wc -l);
C5=$(awk '$5>5' "${Out}"|wc -l);
C10=$(awk '$5>10' "${Out}"|wc -l);
p1=$(printf "%.3f" `echo "scale=3;${C1}*100/${target_length}"|bc`);
p5=$(printf "%.3f" `echo "scale=3;${C5}*100/${target_length}"|bc`);
p10=$(printf "%.3f" `echo "scale=3;${C10}*100/${target_length}"|bc`);
printf "%-s\t %.3f\t %.3f\t %.3f\t %.3f\n" $id $p1 $av $p5 $p10>>${id}.coverage
```
# python script to calculate heterozigosity 
> input:individual vcf file for each sample(before assemble)
```python
import os
import re
sample_list = "sample_list" 
output_file = "het.txt"
txt_out = open(output_file,'a')
txt_out.write("id\tnSNPs\tnHeter_SNP\n")
with open(sample_list,'r') as id_list:
    for id in id_list:
        id = id.strip("\n")
        vcf_file = id + ".vcf"
        nSNP = 0
        nHeter_SNP = 0
        with open(vcf_file,'r') as vcf_lines:
            for line in vcf_lines:
                if line.startswith("NC_007605"):
                    if "INDEL" not in line:
                        nSNP += 1
                        if "1/1" not in line:
                            AD_list = "".join(re.findall(r"^.+AD=(.+?);.+$",line)).split(",")
                            AD_list = [ int(x) for x in AD_list ]
                            AD_list.sort(reverse = True)
                            AD = 0
                            for AD_number in AD_list:
                                AD += int(AD_number)
                            if (float(AD_list[1]) >= 5) and ((float(AD_list[1])/float(AD)) >= 0.1):
                                nHeter_SNP += 1
                            if len(AD_list) > 2:
                                print id + " " + line
        txt_out.write(id + "\t" + str(nSNP) + "\t" + str(nHeter_SNP) + "\n")
```
